### PR TITLE
max retention log file num

### PR DESCRIPTION
**Pull request by DoraALin**
support max retention log file num by `*loggingT#SetMaxRetentionLogNum`

log worker in background clean oldest log files when log file number exceeds max retention file num.

### CODE REVIEW

File: glog_test.go [hunk 1]
```diff
@@ -531,3 +535,31 @@ func BenchmarkUseFsync(b *testing.B) {
...
+	StartWorker(time.Second)
+	innerlogging.file = [numSeverity]flushSyncWriter{}
+	innerlogging.createFiles(fatalLog)
+	for i := 0; i < 2; i++ {
+		for _, file := range innerlogging.file {
+			file.(*syncBuffer).rotateFile(time.Now())
+		}
+	}
+	//make sure there are only 2 file after 1s
+	time.Sleep(3 * time.Second)
```
Comment (absolute8511) [inline glog_test.go]: Pls add more test cases for different retention number

Comment (DoraALin) [reply glog_test.go]: will do

**Comment by DoraALin** [comment]
@absolute8511 different retention file num in testcase

**Comment by absolute8511** [comment]
LGTM

**Pull request merged**